tests/all/component_model/call_hook.rs

```diff
@@ -70,6 +71,87 @@ fn call_wrapped_func() -> Result<()> {
     Ok(())
 }
 
+// Crate a synchronous Func, call it directly:
+#[test]
+fn call_wrapped_func_with_realloc() -> Result<()> {
+    let wat = format!(
+        r#"
+        (component
+            (import "f" (func $f (param "list" (list u8))))
+
+            (core module $libc
+              (memory (export "memory") 1)
+              {REALLOC_AND_FREE}
+            )
+            (core instance $libc (instantiate $libc))
+
+            (core func $f_lower
+                (canon lower (func $f) (memory $libc "memory"))
+            )
+            (core module $m
+                (import "" "" (func $f (param i32 i32)))
+
+                (memory (export "memory") 1)
+
+                (func $export (param i32) (param i32)
+                    (call $f (local.get 0) (local.get 1))
+                )
+
+                (export "export" (func $export))
+
+                (func (export "realloc") (param i32 i32 i32 i32) (result i32)
+                    i32.const 65532)
```

Comment:
Instead of defining this here you can also use the `"realloc"` export of the `$libc` module